ChangeLog.d/rsa-mutex.txt

```diff
@@ -0,0 +1,5 @@
+Bugfix
+   * Ensure that calling mbedtls_rsa_free() twice is safe. This happens
+     when some Mbed TLS library functions fail. Such a double-free was safe
+     when MBEDTLS_THREADING_C was enabled on platforms where freeing a mutex
+     twice is not safe.
```

Comment:
The last sentence is not clear (and I think it says the opposite of what it should). How about something more concrete like "This fixes a potential resource leak when MBEDTLS_THREADING_C is enabled and the mutex implementation allocates resources on mutex initialisation."?

Actually perhaps the entry should start with that: what are we doing that users care about? Fixing a potential resource leak in RSA with threading. When could that leak happen? On some error paths when init-ing a mutex allocates resources. How are we doing it? By ensuring it's safe to call `rsa_free()` twice.

Reply:
There was no resource leak in RSA. The issue was a double free. (In testing terms, the DRBG tests revealed more init calls than free, whereas the RSA tests revealed more free calls than init.) So I think the description is accurate.

The last sentence was indeed missing a negation, so I'm fixing that.

Reply:
Oh, sorry, my bad, I mixed up the two cases indeed.

Reply:
Actually there were two bugs in RSA: a double free and a leak. I modified the commit that adds the changelog entry to add another entry for the leak.